### DISCUSSION TITLE
ERROR in Fundme : can't proceed deploy

### DISCUSSION THREAD

**Discussion by Giri-nandha-57** [Cyfrin/foundry-full-course-cu#1655] [Q&A]
getting this msg/error while deploying my fundme code.

**creation of FundMe pending...

You may want to cautiously increase the gas limit if the transaction went out of gas.**

`// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Getpricefeeds} from "./LibraryPriceFeed.sol"; 

using Getpricefeeds for uint;

contract FundMe {
    
    uint public minimumUSD = 5e18;
    address[] public Funders;
    mapping (address => uint) public FundersDict;

    address public owner;
    constructor() {
        owner = msg.sender;
    }

    function Fund () public payable {
        require(msg.value.GetConversion() >= minimumUSD," Not enough ETH to continue!! ");
        Funders.push(msg.sender);
        FundersDict[msg.sender] += msg.value ;
    } 
    function withdraw () public OnlyOwner {
        for(uint FundersIndex = 0; FundersIndex < Funders.length; FundersIndex++){
            address Funder = Funders[FundersIndex];
            FundersDict[Funder] = 0;
        }
        Funders = new address[](0);
        // 1) using transfer
        // payable(msg.sender).transfer(address(this).balance);
        // 2) using send  ---> returns BOOL
        // SendStatus = payable(msg.sender).send(address(this).balance);
        // require(SendStatus, "Transaction failed!!")
        // 3) using call  ---> returns the called function return
        (bool Callstatus,) = payable(owner).call{value: address(this).balance}("");
        require(Callstatus, "Call Failed!!");        
    }

    modifier OnlyOwner {
        require(owner == msg.sender);
        _;
    }
}
`

**Comment by akashnikam25** [comment]
@Giri-nandha-57  where are you running this program? 
I tried on remix ide it is working fine. I have used metamask as an environment. 
<img width="1064" alt="Screenshot 2024-04-28 at 7 22 29 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/110417279/48f0f55c-227b-461e-b910-c9f8807e2e72">

**Comment by EngrPips** [comment]
```javascript
`// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Getpricefeeds} from "./LibraryPriceFeed.sol";

using Getpricefeeds for uint;

contract FundMe {

uint public minimumUSD = 5e18;
address[] public Funders;
mapping (address => uint) public FundersDict;

address public owner;
constructor() {
    owner = msg.sender;
}

function Fund () public payable {
    require(msg.value.GetConversion() >= minimumUSD," Not enough ETH to continue!! ");
    Funders.push(msg.sender);
    FundersDict[msg.sender] += msg.value ;
} 
function withdraw () public OnlyOwner {
    for(uint FundersIndex = 0; FundersIndex < Funders.length; FundersIndex++){
        address Funder = Funders[FundersIndex];
        FundersDict[Funder] = 0;
    }
    Funders = new address[](0);
    // 1) using transfer
    // payable(msg.sender).transfer(address(this).balance);
    // 2) using send  ---> returns BOOL
    // SendStatus = payable(msg.sender).send(address(this).balance);
    // require(SendStatus, "Transaction failed!!")
    // 3) using call  ---> returns the called function return
    (bool Callstatus,) = payable(owner).call{value: address(this).balance}("");
    require(Callstatus, "Call Failed!!");        
}

modifier OnlyOwner {
    require(owner == msg.sender);
    _;
}
}
```

**Reply by EngrPips** [reply]
Hello @ Giri-nandha-57, Providing more details about your deployment process and the full output of the deployment might help us to know more and suggest a solution.
